ocaml/xapi/xapi_vdi_helpers.ml

```diff
@@ -23,6 +23,87 @@ open Threadext
 module D=Debug.Make(struct let name="xapi" end)
 open D
 
+(* Attach the rrd-stats vdi to dom0 and get the device *)
+let get_rrd_vdi_device ~__context ~vdi =
+	let open Db_filter_types in
+	let vbds = Db.VBD.get_refs_where ~__context ~expr:(And (
+		Eq (Field "VDI", Literal (Ref.string_of vdi)),
+		Eq (Field "currently_attached", Literal "true"))
+	) in
+	let (vbd, device) =
+		match vbds with
+		| vbd :: _ -> (vbd, Db.VBD.get_device ~__context ~self:vbd)
```

Comment:
You should probably check the RRD's magic number here and fail if it is not correct.